src/zope/meta/setup_to_pyproject.py

```diff
@@ -0,0 +1,448 @@
+                        except ValueError:
+                            # Probably a variable in the list
+                            lst = []
+                            for member in value.elts:
+                                if isinstance(member,
+                                              (ast.Constant,
+                                               ast.List,
+                                               ast.Tuple)):
+                                    lst.append(ast.literal_eval(member.value))
+                                elif isinstance(member, ast.BinOp):
+                                    unformatted = member.left.value
+                                    variable = assigned_names.get(
+                                        member.right.id, '')
+                                    formatted = unformatted.replace(
+                                        '%s', variable)
+                                    lst.append(formatted)
+                                else:
+                                    lst.append(ast.literal_eval(member.value))
+                            gathered[key.value] = lst
+                    else:
+                        try:
+                            gathered[key.value] = ast.literal_eval(value)
+                        except ValueError:
+                            print('XXX Cannot convert dictionary value XXX')
+                            print('XXX Please fix setup.py manually first XXX')
+                            print(f'XXX Dictionary key: {key.value} XXX')
+                            print(ast.dump(value, indent=2))
+                            sys.exit(1)
+                setup_kwargs[kw_arg.arg] = gathered
+        elif isinstance(kw_arg.value, ast.Name):
+            if kw_arg.value.id in assigned_names:
+                value = assigned_names.get(kw_arg.value.id)
+            else:
+                value = kw_arg.value.id
+            setup_kwargs[kw_arg.arg] = value
+
+    return setup_kwargs
+
+
+def setup_args_to_toml_dict(setup_kwargs):
+    """ Iterate over setup_kwargs and generate a dictionary of values suitable
+    for pyproject.toml and a dictionary with unconverted arguments
+    """
+    toml_dict = {'project': {}}
+    p_data = toml_dict['project']
+
+    for key in IGNORE_KEYS:
+        setup_kwargs.pop(key, None)
+
+    for key in UNCONVERTIBLE_KEYS:
+        setup_kwargs.pop(key, None)
+
+    for key in PROJECT_SIMPLE_KEYS:
+        if key in setup_kwargs:
+            p_data[key] = setup_kwargs.pop(key)
+
+    license = setup_kwargs.pop('license', 'ZPL-2.1')
+    license.replace('ZPL 2.1', 'ZPL-2.1')
+    p_data['license'] = license
+
+    classifiers = setup_kwargs.pop('classifiers', [])
+    new_classifiers = []
+    for classifier in classifiers:
+        if classifier.startswith('License'):
+            continue
+        elif classifier == 'Framework :: Zope2':
+            new_classifiers.append('Framework :: Zope :: 2')
```

Comment:
I'd suggest to omit this classifier: I doubt that any of our packages is still compatible with Zope 2.

Reply:
The code now leaves out both of those styles for Zope 2.